### ISSUE TITLE
azure_mgmt_web: Error("invalid type: null, expected a sequence", line: 1, column: 994)

### ISSUE THREAD

**Issue by sycured** [Azure/azure-sdk-for-rust#1113]
Hello,

I try to get a web app configuration in JSON but still got the error:
```
Error: Error { context: Custom(Custom { kind: DataConversion, error: Error("invalid type: null, expected a sequence", line: 1, column: 994) }) }
```

Code to reproduce:
```rust
use azure_identity::AzureCliCredential;
use azure_mgmt_web::ClientBuilder;
use serde_json::{to_string_pretty, Value};
use std::{error::Error, sync::Arc};

#[tokio::main]
async fn main()-> Result<(), Box<dyn Error>> {
    let resource_group = "";
    let webapp_name = "";
    let sub_id = "";

    let creds = Arc::new(AzureCliCredential::default());
    let client = ClientBuilder::new(creds).build();

    let get = client.web_apps_client().get_configuration(resource_group, webapp_name, sub_id).into_future().await?;
    let data = to_string_pretty(&get.properties.unwrap())?;
    println!("{:?}", data);

    Ok(())
}
```
---
I also tried to convert to Value as an intermediate step and got another error:
```
the trait bound `Value: From<SiteConfig>` is not satisfied
```

Code to reproduce:
```rust
use azure_identity::AzureCliCredential;
use azure_mgmt_web::ClientBuilder;
use serde_json::{to_string_pretty, Value, to_vec_pretty};
use std::{error::Error, sync::Arc};

#[tokio::main]
async fn main()-> Result<(), Box<dyn Error>> {
    let resource_group = "";
    let webapp_name = "";
    let sub_id = "";

    let creds = Arc::new(AzureCliCredential::default());
    let client = ClientBuilder::new(creds).build();

    let config = client.web_apps_client().get_configuration(resource_group, webapp_name, sub_id).into_future().await?;
    let config_json: Value = config.properties.unwrap().into();
    println!("{:?}", to_string_pretty(&config_json).unwrap());

    Ok(())
}
```

**Comment by cataggar** [comment]
I can reproduce with the first example. Here is the returned json with some identifiers changed:

``` json
{
  "id": "/subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/myrg/providers/Microsoft.Web/sites/mysite/config/web",
  "name": "mysite",
  "type": "Microsoft.Web/sites/config",
  "location": "East US",
  "properties": {
    "numberOfWorkers": 1,
    "defaultDocuments": [
      "Default.htm",
      "Default.html",
      "Default.asp",
      "index.htm",
      "index.html",
      "iisstart.htm",
      "default.aspx",
      "index.php",
      "hostingstart.html"
    ],
    "netFrameworkVersion": "v4.0",
    "phpVersion": "",
    "pythonVersion": "",
    "nodeVersion": "",
    "powerShellVersion": "",
    "linuxFxVersion": "DOCKER|myacr.azurecr.io/myapp:7.2.5210-37335542",
    "windowsFxVersion": null,
    "requestTracingEnabled": true,
    "requestTracingExpirationTime": "9999-12-31T23:59:00Z",
    "remoteDebuggingEnabled": false,
    "remoteDebuggingVersion": null,
    "httpLoggingEnabled": true,
    "azureMonitorLogCategories": null,
    "acrUseManagedIdentityCreds": true,
    "acrUserManagedIdentityID": null,
    "logsDirectorySizeLimit": 40,
    "detailedErrorLoggingEnabled": true,
    "publishingUsername": "$mysite",
    "publishingPassword": null,
    "appSettings": null,
    "metadata": null,
    "connectionStrings": null,
    "machineKey": null,
    "handlerMappings": null,
    "documentRoot": null,
    "scmType": "None",
    "use32BitWorkerProcess": true,
    "webSocketsEnabled": false,
    "alwaysOn": true,
    "javaVersion": null,
    "javaContainer": null,
    "javaContainerVersion": null,
    "appCommandLine": "",
    "managedPipelineMode": "Integrated",
    "virtualApplications": [
      {
        "virtualPath": "/",
        "physicalPath": "site\\wwwroot",
        "preloadEnabled": true,
        "virtualDirectories": null
      }
    ],
    "winAuthAdminState": 0,
    "winAuthTenantState": 0,
    "customAppPoolIdentityAdminState": false,
    "customAppPoolIdentityTenantState": false,
    "runtimeADUser": null,
    "runtimeADUserPassword": null,
    "loadBalancing": "LeastRequests",
    "routingRules": [],
    "experiments": { "rampUpRules": [] },
    "limits": null,
    "autoHealEnabled": false,
    "autoHealRules": null,
    "tracingOptions": null,
    "vnetName": "",
    "vnetRouteAllEnabled": false,
    "vnetPrivatePortsCount": 0,
    "publicNetworkAccess": null,
    "siteAuthEnabled": true,
    "siteAuthSettingsV2": {
      "globalValidation": {
        "requireAuthentication": true,
        "unauthenticatedClientAction": "Return401"
      },
      "identityProviders": {
        "azureActiveDirectory": {
          "enabled": true,
          "registration": {
            "clientId": "00000000-0000-0000-0000-00000000000",
            "clientSecretSettingName": "APP_REGISTRATION_SECRET"
          },
          "validation": {
            "allowedAudiences": ["00000000-0000-0000-0000-00000000000"]
          }
        }
      }
    },
    "cors": null,
    "push": null,
    "apiDefinition": null,
    "apiManagementConfig": null,
    "autoSwapSlotName": null,
    "localMySqlEnabled": false,
    "managedServiceIdentityId": 3006,
    "xManagedServiceIdentityId": null,
    "keyVaultReferenceIdentity": null,
    "ipSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 2147483647,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "ipSecurityRestrictionsDefaultAction": null,
    "scmIpSecurityRestrictions": [
      {
        "ipAddress": "Any",
        "action": "Allow",
        "priority": 2147483647,
        "name": "Allow all",
        "description": "Allow all access"
      }
    ],
    "scmIpSecurityRestrictionsDefaultAction": null,
    "scmIpSecurityRestrictionsUseMain": false,
    "http20Enabled": true,
    "minTlsVersion": "1.2",
    "minTlsCipherSuite": null,
    "supportedTlsCipherSuites": null,
    "scmMinTlsVersion": "1.0",
    "ftpsState": "Disabled",
    "preWarmedInstanceCount": 0,
    "functionAppScaleLimit": null,
    "elasticWebAppScaleLimit": 0,
    "healthCheckPath": "/api/health",
    "fileChangeAuditEnabled": false,
    "functionsRuntimeScaleMonitoringEnabled": false,
    "websiteTimeZone": null,
    "minimumElasticInstanceCount": 1,
    "azureStorageAccounts": {},
    "http20ProxyFlag": 0,
    "sitePort": null,
    "antivirusScanEnabled": false,
    "storageType": "StorageVolume"
  }
}
```

**Comment by cataggar** [comment]
It is failing where `null` values are being returned where arrays are expected. These four fail:

- "appSettings": null,
- "connectionStrings": null,
- "handlerMappings": null,
- "virtualDirectories": null

Which are mapped like so:

``` rust
#[serde(rename = "appSettings", default, skip_serializing_if = "Vec::is_empty")]
pub app_settings: Vec<NameValuePair>,

#[serde(rename = "connectionStrings", default, skip_serializing_if = "Vec::is_empty")]
pub connection_strings: Vec<ConnStringI

#[serde(rename = "handlerMappings", default, skip_serializing_if = "Vec::is_empty")]
pub handler_mappings: Vec<HandlerMapping>,

#[serde(rename = "virtualDirectories", default, skip_serializing_if = "Vec::is_empty")]
pub virtual_directories: Vec<VirtualDirectory>,
```

The service should not be returning null for arrays:

> For arrays, sending/receiving a null array entry is not supported and should result in an error.
https://azure.github.io/autorest/extensions/#x-nullable

But it is. 

The spec is here:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/CommonDefinitions.json#L3205

Nothing is marked as `x-nullable`. It would be better if the service didn't return nulls.

But it is. I suggest a custom deserializer that turns null into empty for an array. Then like other workarounds, it can be configured in an autorust.toml.

**Comment by cataggar** [comment]
Other workarounds are to change the types to `Option<Vec_>`. If you just wanted to use different models, you could get directly at the bytes:

``` rust
    let get = client.web_apps_client().get_configuration(resource_group, webapp_name, sub_id).send().await?;
    let data = get.into_raw_response().into_body().collect().await?;
    println!("{:?}", data);
  ```

**Comment by cataggar** [comment]
@sycured, this was fixed by #1115.  Thanks for reporting it. You can use it with a Rust git dependency for azure_mgmt_web until the next version is published.
